plugins/module_utils/azure_rm_common.py

```diff
@@ -1440,7 +1445,9 @@ def __init__(self, auth_source=None, profile=None, subscription_id=None, client_
             cloud_environment=cloud_environment,
             cert_validation_mode=cert_validation_mode,
             api_profile=api_profile,
-            adfs_authority_url=adfs_authority_url)
+            adfs_authority_url=adfs_authority_url,
+            x509_certificate=x509_certificate,
```

Comment:
@abikouo Thank you very much for your contribution. There is a problem with the direct transmission of this parameter. I tested the path of the certificate locally, and then read it and then pass it. Can you share your certificate format, briefly share how the certificate was created, and the value of the X509_certificate configuration?
```e.g.
            with open(self.credentials.get('certificate_path'), 'rb') as f:
                data = f.read()
            ----------------------
            self.azure_credentials = self.acquire_token_with_client_certificate(
                self._adfs_authority_url,
                self._cloud_environment.endpoints.active_directory_resource_id,
                data,
                self.credentials['thumbprint'],
                self.credentials['client_id'],
                self.credentials['tenant'])

Reply:
@Fred-sun thanks for the feedback.
Do you mean you cannot log in using the ``x509_certificate`` as a parameter? 
I will try to do a brief description on how to use it and add it in the PR description

Reply:
@abikouo However, your change only supports the Track1 SDK, and you also need to support the Track2 SDK! Thank you very much!